sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -2204,4 +2204,37 @@ class CSVSuite extends QueryTest with SharedSparkSession with TestCsvData {
       checkAnswer(resultDF, Row("a", 2, "e", "c"))
     }
   }
+
+  test("filters push down") {
+    Seq(true, false).foreach { multiLine =>
```

Comment:
We don't need to test multiline config because the input doesn't cross-line.

Reply:
Lines in CSV cannot be spitted, so, input should be the same. The difference is how do we read the file - as whole or by lines. But you are right, there should be not difference for the changes since I touched only value conversions.

Reply:
And this is some kind of end-to-end test, I would test as much as possible options and modes.